rock-core/tools-syskit
# fix: race condition in detection of monitor thread death in unmanaged deployments

UnmanagedTasksManager#wait_termination would not detect a monitor
thread that died if the exception happened between the call
to verify_threads_state and the check on dead?

Since verify_threads_state is a no-op if dead? is false, let's
check on dead? first and then call verify_threads_state to check
why the thread died.

This would have zero practical consequence in practice, but
makes some tests fail randomly (albeit very rarely)
